internal/jobservice/application.go

```diff
@@ -30,12 +34,47 @@ func New() *App {
 	return &App{}
 }
 
+var DefaultConfiguration = &configuration.JobServiceConfiguration{
+	GrpcPool: grpcconfig.GrpcPoolConfig{
+		InitialConnections: 5,
+		Capacity:           5,
+	},
+}
+
+// Mutates config where possible to correct mis-configurations.
+// Returns a non-nil error if mis-configuration is unrecoverable.
+func RectifyConfig(config *configuration.JobServiceConfiguration) error {
+	logger := log.WithField("JobService", "RectifyConfig")
+
+	// Grpc Pool
+	if config.GrpcPool.InitialConnections <= 0 {
+		logger.WithFields(log.Fields{
+			"default":    DefaultConfiguration.GrpcPool.InitialConnections,
+			"configured": config.GrpcPool.InitialConnections,
+		}).Warn("config.GrpcPool.InitialConnections invalid, using default instead")
+		config.GrpcPool.InitialConnections = DefaultConfiguration.GrpcPool.InitialConnections
+	}
+	if config.GrpcPool.Capacity <= 0 {
+		logger.WithFields(log.Fields{
+			"default":    DefaultConfiguration.GrpcPool.Capacity,
+			"configured": config.GrpcPool.Capacity,
+		}).Warn("config.GrpcPool.Capacity invalid, using default instead")
+		config.GrpcPool.Capacity = DefaultConfiguration.GrpcPool.Capacity
+	}
+
+	return nil
+}
+
 func (a *App) StartUp(ctx context.Context, config *configuration.JobServiceConfiguration) error {
 	// Setup an errgroup that cancels on any job failing or there being no active jobs.
 	g, _ := errgroup.WithContext(ctx)
 
-	log := log.WithField("JobService", "Startup")
+	err := RectifyConfig(config)
+	if err != nil {
```

Comment:
I think this panic would never be hit.

Reply:
It won't but returning an error from `RectifyConfig` in the future is intended to indicate an unrecoverable error with `config`. Just putting in the correct behavior now.